### ISSUE TITLE
Double cookie is set when running end to end tests with subdomains

### ISSUE THREAD

**Issue by bahmutov** [bahmutov/express-sessions-tutorial#4]
## Testing via tunnel tool

You can use a tunneling tool like [ngrok](https://ngrok.com/) to test the
demo against domain host (which might uncover cookie and session issues).
- Start the `ngrok` tool tunneling local port "3000" - `ngrok http 3000`
  - this shows you an external url, for example `b70f956e.ngrok.io`
- Start local server with `npm start` which runs on port "3000"
  - by default the responses do not set "Domain" for the session or CSRF cookie
  - you can see it by looking at the response headers

``` sh
$ curl -I http://b70f956e.ngrok.io
HTTP/1.1 200 OK
set-cookie: server-session-cookie-id=AAABBB...; Path=/; HttpOnly
```
- Run Cypress tests against the external url passing it via `HOST` variable

``` sh
$ CYPRESS_HOST=http://b70f956e.ngrok.io cypress open
```
- When the tests finish, in the Cypress-controlled browser open the DevTools
  and look at the application's cookies for the domain
  `http://b70f956e.ngrok.io`. You will see 2 session cookies!

```
server-session-cookie-id  AAABBB...   b70f956e.ngrok.io
server-session-cookie-id  AAABBB...   .b70f956e.ngrok.io
```
- Close Cypress application
- Stop local server
- Start the local server again but pass both the external host and the
  "set domain for cookies flag". You don't have to restart `ngrok`

``` sh
$ HOST=b70f956e.ngrok.io DOMAIN=1 npm start
```
- Check that cookies have explicit domain set

``` sh
$ curl -I http://b70f956e.ngrok.io
HTTP/1.1 200 OK
set-cookie: server-session-cookie-id=AAABBB...; Domain=b70f956e.ngrok.io; Path=/; HttpOnly
```
- Start the Cypress application again

``` sh
$ CYPRESS_HOST=http://b70f956e.ngrok.io cypress open
```
- Run the tests, open Chrome DevTools and notice that only single session cookie
  is there

```
server-session-cookie-id  AAABBB...   .b70f956e.ngrok.io
```

**Comment by bahmutov** [comment]
Seems without explicit "Domain" set on a cookie, Cypress sets it for cases when there is full domain (no "localhost") twice: `.domain.com` and `domain.com`. I would really not like to use explicit "Domain" when setting cookies because it will break every use of a proxy in our case.

**Comment by bahmutov** [comment]
Resolved in cypress 0.17.2
